odc/stats/plugins/l34_utils/lc_water_seasonality.py

```diff
@@ -0,0 +1,40 @@
+import xarray as xr
+
+from odc.stats._algebra import expr_eval
+from . import utils
+
+WATER_FREQ_NODATA = -999
+
+
+def water_seasonality(xx: xr.Dataset, water_seasonality_threshold):
+    # >= 3 months ----> 1  Semi-permanent or permanent
+    # < 3 months  ----> 2 Temporary or seasonal
+
+    water_season_mask = expr_eval(
+        "where((a>watseas_trh)&(a<=12), 100, a)",
+        {"a": xx.water_frequency.data},
+        name="mark_water_season",
+        dtype="uint8",
+        **{"watseas_trh": water_seasonality_threshold},
+    )
+    water_season_mask = expr_eval(
+        "where((a<=watseas_trh)&(a<=12), 200, a)",
+        {"a": water_season_mask},
+        name="mark_water_season",
+        dtype="uint8",
+        **{"watseas_trh": water_seasonality_threshold},
+    )
+    water_season_mask = expr_eval(
+        "where((a==watersea_nodata), 255, a)",
+        {"a": water_season_mask},
+        name="mark_water_season",
+        dtype="uint8",
+        **{
+            "watseas_trh": water_seasonality_threshold,
+            "watersea_nodata": WATER_FREQ_NODATA,
```

Comment:
the correct `nodata=255` from the raster, also it won't work as `water_season_mask` has been converted to `uint8` from last step, `-999` was lost had it been the effective `nodata`.

Reply:
Good point, refactored the code to effectively replace nodata"